File: Frontend/src/services/teamService.js
import api from './api';

export const teamService = {
  getAllTeams: async () => {
    const response = await api.get('/teams');
    return response.data;
  },

  getTeamDetails: async (teamId) => {
    const response = await api.get(`/teams/${teamId}`);
    return response.data;
  },

  getTeamsByCaptain: async (captainId) => {
    const response = await api.get(`/teams/captain/${captainId}`);
    return response.data;
  },

  joinTeam: async (teamId, userId, inGameName) => {
    try {
      const response = await api.post(`/teams/${teamId}/join`, { userId, inGameName });
      return response.data;
    } catch (error) {
      throw error;
    }
  },

  approveJoinRequest: async (teamId, memberId, captainId) => {
    const response = await api.put(`/teams/${teamId}/members/${memberId}/approve`, null, {
      params: { captainId }
    });
    return response.data;
  },

  rejectJoinRequest: async (teamId, memberId, captainId) => {
    const response = await api.put(`/teams/${teamId}/members/${memberId}/reject`, null, {
      params: { captainId }
    });
    return response.data;
  },

  kickMember: async (teamId, memberId, captainId) => {
    const response = await api.delete(`/teams/${teamId}/members/${memberId}/kick`, {
      params: { captainId }
    });
    return response.data;
  }
};

File: Backend/src/main/java/com/tournament/engine/modules/tournament/service/TeamService.java
package com.tournament.engine.modules.tournament.service;

import com.tournament.engine.modules.tournament.dto.TeamResponse;

import java.util.List;

public interface TeamService {
    List<TeamResponse> getAllTeams();
    TeamResponse getTeamDetails(Long teamId);
    void joinTeam(Long teamId, com.tournament.engine.modules.tournament.dto.JoinTeamRequest request);
    List<TeamResponse> getTeamsByCaptain(Long captainId);
    void approveJoinRequest(Long teamId, Long memberId, Long captainId);
    void rejectJoinRequest(Long teamId, Long memberId, Long captainId);
    void kickMember(Long teamId, Long memberId, Long captainId);
    
    // Admin features
    void deleteTeam(Long teamId);
}
